docs/spec/instrument/hooks.md

```diff
@@ -1,6 +1,733 @@
+                "organization": {
+                    "id": "d8b0a63e-9a5d-4638-b5a3-4361ba067200",
+                    "name": "Azura"
+                }
+            }
+        }
+    }
+}
+   ```
+
+## 8. Agent Response
+
+### 8.1. Description
+This hook is called when the agent sends back a response.<br>
+This hook should be used **before** the agent response is sent to the user. <br>
+
+### 8.2. Method
+[`steps/message`](specification.md#45-stepsmessage)<br><br>
+This method is used with [User Message](#4-user-message) hook.<br>
+For this hook `role` **MUST** be `agent` (see example).
+
+
+### 8.3. Reponse
+The response is an [`AOSSuccessResponse`](specification.md#51-aossuccessresponse-object) object.
+
+| Decision | Behavior |
+| :--------- | :---------- |
+| `allow` | The agent response should be sent as is. |
+| `deny` | The agent response should be blocked. Recommended to send a response indicating that the original response was blocked. |
+| `modify` | The agent response should be sent back with the modified content found in `modifiedRequest` field. |
+
+
+### 8.4. Example
+   ```json
+{
+    "jsonrpc": "2.0",
+    "method": "steps/message", 
+    "id": "716601aa-36eb-4720-ab08-c59b9321aecb",
+    "params": {
+        "message": {
+            "role": "agent",
+            "id": "a66c132e-a554-4dfc-8a47-2db66e13ef39",
+            "content": [
+                {
+                    "kind": "text",
+                    "text": "The bank account of Acme Corp is 000123456789"
+                }
+            ]
+        },
+        "citations": [
+            {
+                "kind": "file",
+                "id": "0a267158-7b44-452a-bba8-c1107bdf6128",
+                "name": "Bank Accounts.xlsx"
+            }
+        ],
+        "reasoning": "Found Acme Corp bank account details in Bank Accounts.xlsx. I can respond to the user.",
+        "context": {
+            "agent": {
+                "id": "1c88ab7d-395f-449a-af51-6028f9e842ea",
+                "name": "Payments agent",
+                "instructions": "You are very helpful agent. You manage customers bank accounts and payments",
+                "version": "8878",
+                "provider": {
+                    "name": "OpenAI",
+                    "url": "https://openai.com/"
+                }
+            },
+            "session": {
+                "id": "84c36ebb-83aa-4bc9-8670-7aba4cedc70f"
+            },
+            "turnId": "083db36a-5ba1-4d37-8c3f-ebc2ec23b96b", 
+            "stepId": "fdee9786-1754-4c87-962c-a1ed02918b99",
+            "timestamp": "2025-01-24T15:33:45.123Z",
+            "user": {
+                "id": "8cc6e9bc-6ad5-4b95-8060-300915b1aaba",
+                "email": "user@company.io",
+                "organization": {
+                    "id": "d8b0a63e-9a5d-4638-b5a3-4361ba067200",
+                    "name": "Azura"
+                }
+            }
+        }
+    }
+}
+   ```
+
+# A2A protocol hooks
```

Comment:
Shouldn't this be in the A2A section, referencing events here that are being implemented via A2A?

Reply:
done. added all a2a methods to the spec and added relevant hooks in the outbound direction in extend_a2a